application/actions/app.go

```diff
@@ -95,6 +95,7 @@ func App() *buffalo.App {
 		app.POST("/gql/", gqlHandler)
 
 		app.GET("/conversations", usersThreads)
+		app.POST("/markMessagesAsRead", threadsMarkMessagesAsRead)
```

Comment:
Maybe. Do we need to be concerned with sticking to the current gql query spelling or can we ignore that?

Reply:
I agree, I think it'd be better to see all thread related stuff under a common path. I'd expect endpoints something like these, though there may be better names/structures. 

```
// list all my threads
GET /threads

// get specific thread
GET /threads/{id}

// mark thread as read
PUT /threads/{id}/read

// add message to thread
POST /threads/{id}
```

Reply:
> Do we need to be concerned with sticking to the current gql query spelling or can we ignore that?

That can be ignored. That much will have to change in the UI code anyway, since GQL query name doesn't translate to the REST call.